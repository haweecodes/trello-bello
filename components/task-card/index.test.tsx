import { getByPlaceholderText, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TaskCard from './index';

describe('task card', () => {
  it('should render task card', () => {
    const { container } = render(<TaskCard/>);
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have a text area', () => {
    const { getByPlaceholderText } = render(<TaskCard/>);
    const input = getByPlaceholderText('Title...');
    expect(input.tagName).toBe('INPUT')
  })

  it('should have card content', () => {
    const { getByTestId } = render(<TaskCard/>);
    const cardContent = getByTestId('card-content');
    expect(cardContent).toHaveTextContent('Card content')
  })
})