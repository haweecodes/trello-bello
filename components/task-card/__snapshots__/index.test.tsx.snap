// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`task card should render task card 1`] = `
<div
  class="card"
>
  <textarea
    class="cardTitle"
  >
    Title...
  </textarea>
  <div
    class="cardContent"
  >
    Card content Card content Card content Card content Card content Card content Card content
  </div>
  <div
    class="cardAction"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-default ant-btn-block"
      type="button"
    >
      <span>
        Add Task
      </span>
    </button>
  </div>
</div>
`;
